scrapy/settings/__init__.py

```diff
@@ -88,19 +103,48 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def _getcomposite(self, name):
+        # DO NOT USE THIS FUNCTION IN YOUR CUSTOM PROJECTS
+        # It's for internal use in the transition away from the _BASE settings and
+        # will be removed along with _BASE support in a future release
+        basename = name + "_BASE"
+        if basename in self.attributes:
```

Comment:
nitpick: This line should be `if basename in self:` and other similar references could be replaced too.